tests/next_tests/integration_tests/feature_tests/ffront_tests/test_compiled_program.py

```diff
@@ -0,0 +1,693 @@
+# GT4Py - GridTools Framework
+#
+# Copyright (c) 2014-2024, ETH Zurich
+# All rights reserved.
+#
+# Please, refer to the LICENSE file in the root directory.
+# SPDX-License-Identifier: BSD-3-Clause
+
+import numpy as np
+import pytest
+
+from gt4py import next as gtx
+from gt4py.next.ffront.decorator import Program
+from gt4py.next.ffront.fbuiltins import int32, neighbor_sum
+
+from next_tests.integration_tests import cases
+from next_tests.integration_tests.cases import (
+    V2E,
+    cartesian_case,
+    mesh_descriptor,
+    unstructured_case,
+)
+from next_tests.integration_tests.feature_tests.ffront_tests.ffront_test_utils import (
+    MeshDescriptor,
+    exec_alloc_descriptor,
+    simple_mesh,
+    skip_value_mesh,
+)
+
+
+def _always_raise_callable(*args, **kwargs) -> None:
```

Comment:
Don't need to be changed in this PR but I think this is exactly the same as:
```python
Mock(side_effect=AssertionError('This function should never be called.'))
```

Reply:
no, it didn't work. maybe we can fix it together.